File: examples/clients/python/chat_completions.py
import os
import sys
import json
import urllib.request


def main() -> int:
    base_url = os.environ.get("DITTO_BASE_URL", "http://127.0.0.1:8080").rstrip("/")
    token = os.environ.get("DITTO_VK_TOKEN")
    if not token:
        print("missing DITTO_VK_TOKEN", file=sys.stderr)
        return 1

    payload = {
        "model": "gpt-4o-mini",
        "stream": False,
        "messages": [{"role": "user", "content": "Say hello in one sentence."}],
    }

    req = urllib.request.Request(
        f"{base_url}/v1/chat/completions",
        data=json.dumps(payload).encode("utf-8"),
        headers={
            "content-type": "application/json",
            "authorization": f"Bearer {token}",
            "x-request-id": f"py-{int(__import__('time').time() * 1000)}",
        },
        method="POST",
    )

    with urllib.request.urlopen(req) as resp:
        body = resp.read().decode("utf-8")
        print(body)

    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: examples/clients/go/chat_completions.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"os"
	"strings"
	"time"
)

func main() {
	baseURL := os.Getenv("DITTO_BASE_URL")
	if baseURL == "" {
		baseURL = "http://127.0.0.1:8080"
	}
	baseURL = strings.TrimRight(baseURL, "/")
	token := os.Getenv("DITTO_VK_TOKEN")
	if token == "" {
		fmt.Fprintln(os.Stderr, "missing DITTO_VK_TOKEN")
		os.Exit(1)
	}

	payload := map[string]any{
		"model":  "gpt-4o-mini",
		"stream": false,
		"messages": []map[string]any{
			{"role": "user", "content": "Say hello in one sentence."},
		},
	}

	body, err := json.Marshal(payload)
	if err != nil {
		panic(err)
	}

	req, err := http.NewRequest("POST", baseURL+"/v1/chat/completions", bytes.NewReader(body))
	if err != nil {
		panic(err)
	}
	req.Header.Set("content-type", "application/json")
	req.Header.Set("authorization", "Bearer "+token)
	req.Header.Set("x-request-id", fmt.Sprintf("go-%d", time.Now().UnixMilli()))

	resp, err := http.DefaultClient.Do(req)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	respBody, _ := io.ReadAll(resp.Body)
	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
		fmt.Fprintf(os.Stderr, "HTTP %d: %s\n", resp.StatusCode, string(respBody))
		os.Exit(1)
	}
	fmt.Println(string(respBody))
}
